# Solver summary of ceres solver
# see: http://ceres-solver.org/nnls_solving.html#solver-summary

Header header

string BriefReport
string FullReport
bool IsSolutionUsable
#??? minimizer_type
#??? termination_type
string message
float64 initial_cost
float64 final_cost
float64 fixed_cost
IterationSummary[] iterations
int64 num_successful_steps
int64 num_unsuccessful_steps
int64 num_inner_iteration_steps
float64 preprocessor_time_in_seconds
float64 minimizer_time_in_seconds
float64 postprocessor_time_in_seconds
float64 total_time_in_seconds
float64 linear_solver_time_in_seconds
int64 num_linear_solves
float64 residual_evaluation_time_in_seconds
int64 num_residual_evaluations
float64 jacobian_evaluation_time_in_seconds
int64 num_jacobian_evaluations
float64 inner_iteration_time_in_seconds
int64 num_parameter_blocks
int64 num_parameters
int64 num_effective_parameters
int64 num_residual_blocks
int64 num_residuals
int64 num_parameter_blocks_reduced
int64 num_parameters_reduced
int64 num_effective_parameters_reduced
int64 num_residual_blocks_reduced
int64 num_residuals_reduced
int64 num_threads_given
int64 num_threads_used
#??? linear_solver_type_given
#??? linear_solver_type_used
int64[] linear_solver_ordering_given
int64[] linear_solver_ordering_used
string schur_structure_given
string schur_structure_used
bool inner_iterations_given
bool inner_iterations_used
int64[] inner_iteration_ordering_given
int64[] inner_iteration_ordering_used
#??? preconditioner_type_given
#??? preconditioner_type_used
#??? visibility_clustering_type
#??? trust_region_strategy_type
#??? dogleg_type
#??? dense_linear_algebra_library_type
#??? sparse_linear_algebra_library_type
#??? line_search_direction_type
#??? line_search_type
#??? line_search_interpolation_type
#??? nonlinear_conjugate_gradient_type
int64 max_lbfgs_rank